ee/tpmrunner/tpmrunner.go

```diff
@@ -0,0 +1,241 @@
+)
+
+type tpmRunner struct {
+	signer        crypto.Signer
+	signerCreator tpmSignerCreator
+	store         types.GetterSetterDeleter
+	slogger       *slog.Logger
+	interrupt     chan struct{}
+	interrupted   bool
+}
+
+// tpmSignerCreator is an interface for creating and loading TPM signers
+// useful for mocking in tests
+type tpmSignerCreator interface {
+	CreateKey(opts ...tpm.TpmSignerOption) (private []byte, public []byte, err error)
+	New(private, public []byte) (crypto.Signer, error)
+}
+
+// defaultTpmSignerCreator is the default implementation of tpmSignerCreator
+// using the tpm package
+type defaultTpmSignerCreator struct{}
+
+// CreateKey creates a new TPM key
+func (d defaultTpmSignerCreator) CreateKey(opts ...tpm.TpmSignerOption) (private []byte, public []byte, err error) {
+	return tpm.CreateKey()
+}
+
+// New creates a new TPM signer
+func (d defaultTpmSignerCreator) New(private, public []byte) (crypto.Signer, error) {
+	return tpm.New(private, public)
+}
+
+// tpmRunnerOption is a functional option for tpmRunner
+// useful for setting dependencies in tests
+type tpmRunnerOption func(*tpmRunner)
+
+func New(ctx context.Context, slogger *slog.Logger, store types.GetterSetterDeleter, opts ...tpmRunnerOption) (*tpmRunner, error) {
+	_, span := traces.StartSpan(ctx)
+	defer span.End()
+
+	_, _, err := fetchKeyData(store)
+	if err != nil {
+		return nil, err
+	}
+
+	tpmRunner := &tpmRunner{
+		store:         store,
+		slogger:       slogger.With("component", "tpmrunner"),
+		interrupt:     make(chan struct{}),
+		signerCreator: defaultTpmSignerCreator{},
+	}
+
+	for _, opt := range opts {
+		opt(tpmRunner)
+	}
+
+	return tpmRunner, nil
+}
+
+// Public returns the public key of the current console user
+// creating and peristing a new one if needed
+func (tr *tpmRunner) Execute() error {
+	currentRetryInterval, maxRetryInterval := 1*time.Second, 1*time.Minute
+	retryTicker := time.NewTicker(currentRetryInterval)
+	defer retryTicker.Stop()
+
+	for {
+		ctx := context.Background()
+		signer, err := tr.loadOrCreateKeys(ctx)
+		if err != nil {
+			tr.slogger.Log(ctx, slog.LevelError,
+				"creating tpm signer, will retry",
+				"err", err,
+			)
+
+			if currentRetryInterval < maxRetryInterval {
+				currentRetryInterval += time.Second
+				retryTicker.Reset(currentRetryInterval)
+			}
+		} else {
+			tr.signer = signer
+			retryTicker.Stop()
+		}
+
+		select {
+		case <-retryTicker.C:
+			continue
+		case <-tr.interrupt:
+			tr.slogger.Log(ctx, slog.LevelDebug,
+				"interrupt received, exiting secure enclave signer execute loop",
+			)
+			return nil
+		}
+	}
+}
+
+func (tr *tpmRunner) Interrupt(_ error) {
+	// Only perform shutdown tasks on first call to interrupt -- no need to repeat on potential extra calls.
+	if tr.interrupted {
+		return
+	}
+
+	tr.interrupted = true
+
+	// Tell the execute loop to stop checking, and exit
+	tr.interrupt <- struct{}{}
+}
+
+// Public returns the public key of the current console user
```

Comment:
Is this actually connected to the console user?

Reply:
no, that was just copy pasta, fixed